panther_gazebo/include/panther_gazebo/light_converter_node.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "image_transport/image_transport.hpp"
+#include "ros_gz_interfaces/msg/light.hpp"
+#include "sensor_msgs/msg/image.hpp"
+
+using namespace std::chrono_literals;
+
+namespace panther_gazebo
+{
+
+using ImageMsg = sensor_msgs::msg::Image;
+
+class GZLightConverter : public rclcpp::Node
+{
+public:
+  GZLightConverter(
+    const std::string & node_name, const rclcpp::NodeOptions & options = rclcpp::NodeOptions());
+  void Initialize();
+
+private:
+  void FrameCB(const ImageMsg::ConstSharedPtr msg, std::string light_name_);
+  std::tuple<float, float, float, float> calculateMeanRGBA(
+    const std::vector<unsigned char> & rgbaData);
```

Comment:
```suggestion
  RGBAColor CalculateMeanRGBA(
    const std::vector<std::uint8_t> & rgba_data);
```
We can expect data to have uint8_t type